README.md

```diff
@@ -1,328 +1,526 @@
+```
+
+### Parsing templated variables
+
+Additional variables provided in the Github event [context][event-context] and
+event [payload][event-payload] can be used to replace templated variables in the
+input payload with the `payload-templated` option:
+
 ```yaml
 - name: Send custom JSON data to Slack workflow
   id: slack
-  uses: slackapi/slack-github-action@v1.27.0
+  uses: slackapi/slack-github-action@v2-development
   with:
-    # For posting a rich message using Block Kit
-    payload: |
-      {
-        "text": "GitHub Action build result: ${{ job.status }}\n${{ github.event.pull_request.html_url || github.event.head_commit.url }}",
-        "blocks": [
-          {
-            "type": "section",
-            "text": {
-              "type": "mrkdwn",
-              "text": "GitHub Action build result: ${{ job.status }}\n${{ github.event.pull_request.html_url || github.event.head_commit.url }}"
-            }
-          }
-        ]
-      }
-  env:
-    SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
-    SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
+    payload-file-path: "./payload-slack-content.json"
+    payload-templated: true
+    webhook: ${{ secrets.SLACK_WEBHOOK_URL }}
+    webhook-type: webhook-trigger
 ```
 
-### HTTPS Proxy
+This replaces variables templated as `${{ github.payload.repository.html_url }}`
+with the values found in the GitHub Action event [payload][event-payload].
 
-If you need to use a proxy to connect with Slack, you can use the `HTTPS_PROXY` (or `https_proxy`) environment variable. In this example we use the Slack App technique, but configuring a proxy works the same way for all of them:
+### Proxying HTTPS requests
+
+If you need to use a proxy to connect to Slack, you can use the `proxy` option.
+In this example we use the technique that calls a Slack API method, but
+configuring a proxy is the same for all techniques:
 
 ```yaml
 - name: Post to a Slack channel via a proxy
-  id: slack
-  uses: slackapi/slack-github-action@v1.27.0
+  uses: slackapi/slack-github-action@v2-development
   with:
-    channel-id: 'CHANNEL_ID'
-    slack-message: 'This message was sent through a proxy'
-  env:
-    SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
-    # Set the HTTPS_PROXY environment variable to whatever your policy requires
-    HTTPS_PROXY: 'http://proxy.example.org:8080'
+    method: chat.postMessage
+    proxy: "http://proxy.example.org:8080" # Change this to a custom value
+    token: ${{ secrets.SLACK_BOT_TOKEN }}
+    payload: |
+      channel: ${{ secrets.SLACK_CHANNEL_ID }}
+      text: "This message was sent through a proxy"
 ```
 
-## Contributing
+The `proxy` option can also be provided with the `HTTPS_PROXY` or `https_proxy`
+[environment variable][github-environment] from within the GitHub Actions step.
+
+### Retrying failed requests
+
+Sometimes outgoing requests fail due to [rate limits][rate-limits] or similar
+[HTTP responses][retry-after] and can be retried later.
```

Comment:
Added [a footer to this section](https://github.com/slackapi/slack-github-action/tree/v2-development?tab=readme-ov-file#retrying-failed-requests) with some related details! Some changes to wording and omitted details felt needed since `@slack/web-api` retries are used when sending Slack API methods and `axios-retry` is used with webhooks 🔗

Always open to more updates around here, but for now I'll mark this as **resolved**!